pkg/provider/loadBalancer.go

```diff
@@ -3,7 +3,9 @@ package provider
 import (
 	"context"
 	"fmt"
+	"github.com/billryan/collections/set"
```

Comment:
In golang, there is no existing set, but you can still use built-in map to achieve same functionality, 
`map[string]struct{}` for example. We prefer `map[string]struct{}` than `map[string]bool` because struct{} doesn't consume space https://www.reddit.com/r/golang/comments/qzha6o/for_sets_maptstruct_vs_maptbool/

In general, unless its' very necessary, we tries to avoid importing more external dependencies, since it also introduce the cost of maintaining them, making sure they are up to date, patching CVEs/bugs quickly. So if you could just use `map[string]struct{}` that would be great.